src/transaction_builder.js

```diff
@@ -210,8 +213,18 @@ function prepareInput (input, kpPubKey, redeemScript, hashType) {
     // pay-to-scriptHash is not possible without a redeemScript
     if (input.prevOutType === 'scripthash') throw new Error('PrevOutScript is P2SH, missing redeemScript')
 
+    if (!input.pubKeys || input.pubKeys.length === 0) {
+      // maybe we can derive more information about the script
+      // now that we have a kpPubKey?
+      expanded = expandOutput(input.prevOutScript, input.prevOutType, kpPubKey)
+      input.pubKeys = expanded.pubKeys
+      input.signatures = expanded.signatures
+    }
+
     // throw if we can't sign with it
-    if (!input.pubKeys || !input.signatures) throw new Error(input.prevOutType + ' not supported')
+    if (!input.pubKeys || !input.signatures) {
+      throw new Error(input.prevOutType + ' not supported')
```

Comment:
[If `input.pubKeys.length === 0` by this point](https://github.com/bitcoinjs/bitcoinjs-lib/pull/645/files#r82109281... maybe we should throw `not supported`... though,  I think we need more tests cases.
Either way it will fail anyway, so its as safe either way, just a matter of terminology.